kettle/make_json.py

```diff
@@ -182,24 +181,28 @@ def parse_result(child_node):
         skipped = child_node.findall('skipped')
         return time, failure_text, skipped
 
-    if tree.tag == 'testsuite':
-        for child in tree.findall('testcase'):
-            name = child.attrib['name']
-            time, failure_text, skipped = parse_result(child)
-            if skipped:
-                continue
-            yield make_result(name, time, failure_text)
-    elif tree.tag == 'testsuites':
-        for testsuite in tree:
-            suite_name = testsuite.attrib.get('name', 'unknown')
-            for child in testsuite.findall('testcase'):
-                name = '%s %s' % (suite_name, child.attrib['name'])
+    try:
+        if tree.tag == 'testsuite':
```

Comment:
The name attribute validation needs to be here instead of within the `testsuites` case if you want to be as strict as [the junit5 xsd](https://github.com/junit-team/junit5/blob/main/platform-tests/src/test/resources/jenkins-junit.xsd) which has `name` as a required attribute

My read of what is in this PR is:
- if I pass in a bare `<testsuite>` I don't need  `name` attribute (but it's worth noting [testgrid's junit parsing](https://github.com/GoogleCloudPlatform/testgrid/blob/481a3cbf271b6b36e32113024c6b7c3162bd66c2/metadata/junit/junit.go) assumes `<testsuites>` as the root element)
- if I pass in a `<testsuites><testsuite>...` I do need a `name` attribute on the testsuite

Reply:
I changed it to always make a try for name and default to unspecified.